### PR TITLE
add support for keys that implement Borrow

### PR DESCRIPTION
**Pull request by demoray**
This enables using String as a key, and fetching it with a &str

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//! [`ExpiringMap`] is a wrapper around [`AHashMap`] that allows the specification
+//! [`ExpiringMap`] is a wrapper around [`HashMap`] that allows the specification
```
Comment (randomairborne) [inline src/lib.rs]: Sorry to nit these- Yes, they are wrong. Do you mind fixing them in another PR? I could also do it if you don't want to.

**Review by randomairborne — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (demoray) [reply src/lib.rs]: Done

**Review by randomairborne — APPROVED** [review]
looks good to me
